fontra2fontir/src/fontra.rs

```diff
@@ -3,21 +3,105 @@
 //! See <https://github.com/googlefonts/fontra/blob/main/src/fontra/core/classes.py>
 
 use std::{
-    collections::HashMap,
+    collections::{BTreeMap, HashMap},
     fs,
     path::{Path, PathBuf},
 };
 
 use fontdrasil::types::GlyphName;
 use fontir::error::Error;
-use percent_encoding::{utf8_percent_encode, NON_ALPHANUMERIC};
 use serde::Deserialize;
 use write_fonts::types::Tag;
 
+const SEPARATOR_CHAR: char = '^';
+
+fn is_reserved_char(c: char) -> bool {
+    let c = c as u32;
+    match c {
+        _ if c < 32 => true,
+        _ if c == 0x7F => true,
+        _ if c == SEPARATOR_CHAR as u32 => true,
+        _ => false,
+    }
```

Comment:
A bit more idiomatic to write this as,

```suggestion
        match c {
        '\0' ..= '\x1F' => true,
        SEPARATOR_CHAR => true,
        '\x7F' => true,
        _ => false,
    }
```

and you could condense it down to a once-line `matches!` statement by combining the matches using `|`, like,


```rust
    matches!(c, '\0'..='\x1F' | SEPARATOR_CHAR | '\x7F')
```

Reply:
Much nicer, will update